Upgrading Agent Docs Changes (#29992)
DataDog/documentation: content/en/agent/guide/upgrade_between_minor_versions.md

---
title: Upgrade your Datadog Agent
aliases:
  - /agent/faq/upgrade-between-agent-minor-versions
  - /agent/guide/upgrade-between-agent-minor-versions
  - /agent/versions/upgrade_between_agent_minor_versions/
  - /agent/guide/upgrade
further_reading:
  - link: "/agent/troubleshooting/"
    tag: "Documentation"
    text: "Agent Troubleshooting"
  - link: "/agent/basic_agent_usage/"
    tag: "Documentation"
    text: "Basic Agent Usage"
---

## Overview
Datadog recommends you update your Datadog Agent with every [minor and patch][6] release. This guide walks you through how to roll out a new Agent version across your hosts in a few clicks.

## Upgrade between minor versions of the Agent
### Upgrade remotely with Fleet Automation

> Fleet Automation allows you to upgrade Datadog Agents across non-containerized Linux and Windows environments remotely. In a few clicks you can deploy the latest Agent release, keeping your fleet current with new features and security patches. To get setup, follow the [Fleet Automation Agent Upgrade guide][7].

Reviewer: ```suggestion
[Fleet Automation](/agent/fleet_automation/remote_management) enables you to centrally manage your fleet of Datadog Agents. Fleet Automation includes [Remote Agent Management][2], which allows you to remotely upgrade Agents across non-containerized Linux and Windows environments. 

To upgrade your Agents with Fleet Automation:
1. Enable [Remote Configuration](/agent/remote_config/).
2. Enable [Remote Agent Management][3].
3. [Upgrade your Datadog Agents][7].
```

If I think about how people are first coming to this upgrade guide, I suspect a lot of them don't know anything about Fleet Automation or Remote Agent Management, how those concepts are related, or why they'd want to use them. They also may not know whether they want to upgrade remotely or with a script, or how to go about making that decision.

So for this section, I want the first sentence to introduce what Fleet Automation is and what it's good for (managing multiple Agents at once). Then, in the second sentence, I introduce a second concept: Fleet Automation has a component called Remote Agent Management, which is the part that allows users to do the upgrading of Agents.

When a user clicks the "Fleet Automation Agent Upgrade guide" link, they're brought to a list of instructions where the first step is "Enable Remote Agent Management." They click on that, and they're brought to a list of instructions where the first step has them enable Remote Configuration, and they have to click again to do that first. Going backwards through steps isn't a great experience, so I've listed those out clearly here and in the order they should be taken.

**Note**: I've included some links here that you'll probably want to re-format as numbered links.

### Upgrade with script or configuration management tooling

> Follow the [in-app instructions][4] to upgrade Datadog Agents across container, host-based, and Infrastructure as Code (IaC) tool-managed environments. The guided flow generates an Agent installation command tailored to your platform for upgrading the Agent. By default, the command installs the latest version of the Agent. To upgrade to a specific minor version, set `DD_AGENT_MINOR_VERSION=<TARGET_MINOR>` before running the script. 

Reviewer: This "in-app instructions" link goes to the Fleet Automation page in Datadog. Is that correct? I would assume that this second section pertains to people who are not using Fleet Automation. It kind of looks like this paragraph should go in the previous section, not under this heading.

Author: hmmm I see what you mean...

Author: I think the case here is that the UI has changed to have Fleet Automation for general Agent updates. Remote management is an additional feature that can be used to upgrade agents remotely and the scripts can still be used instead of the remote management. Is this correct, @ethandebnath7702 ?

> ## Upgrade between minor versions of the Agent

Reviewer: What's this next section supposed to be? Looks like there are two headings that say "Upgrade between minor versions of the Agent". I think this section is actually supposed to have the heading "Upgrade with script or configuration management tooling".. right?

Reviewer 2: You're right we should delete line 30 here. This section is already there. CC: @iadjivon

{{< tabs >}}
{{% tab "Linux" %}}

> To upgrade between minor versions of the Agent is to use the `install_script_agent7.sh` script. The following commands work on all supported Linux distributions.

Reviewer: ```suggestion
To upgrade between minor versions of the Agent, use the `install_script_agent7.sh` script. The following commands work on all supported Linux distributions.
```

{{% /tab %}}
{{% tab "MacOS" %}}

> To upgrade between minor version of the Agent is to use the `install_mac_os.sh` script. The following commands work on all supported MacOS versions.

Reviewer: ```suggestion
To upgrade between minor version of the Agent, use the `install_mac_os.sh` script. The following commands work on all supported MacOS versions.
```

Reviewer 2: LGTM

{{% /tab %}}
{{< /tabs >}}

## Upgrade between major versions of the Agent

> ### Upgrade to Datadog Agent 7 

Reviewer: The hierarchy of these headings seems unnecessarily complicated. Can we just have two headings, "upgrade from 6 to 7" and "upgrade from 5 to 7"?

<div class="alert alert-info">

> Agent 7 only supports Python 3 custom checks. <a href="/agent/guide/python-3">Check if your custom checks are Python 3 compatible</a> before upgrading to Agent 7.

Reviewer: ```suggestion
Agent 7 only supports Python 3 custom checks. <a href="/agent/guide/python-3">Verify if your custom checks are Python 3 compatible</a> before upgrading to Agent 7.
```

small nit, I just don't like the phrase "check if your checks"
